packages/amplify-storage-simulator/src/server/S3server.ts

```diff
@@ -248,19 +248,28 @@ export class StorageServer {
       join(String(this.localDirectoryPath), String(request.params.path))
     );
     if (request.query.uploads !== undefined) {
-      this.uploadId = uuid();
+      let id = uuid();
+      this.uploadId.push(id);
+      this.upload_bufferMap[id] = new Object();
       //response.set('Content-Type', 'text/xml');
       response.send(
         o2x({
           '?xml version="1.0" encoding="utf-8"?': null,
           InitiateMultipartUploadResult: {
             Bucket: this.route,
             Key: request.params.path,
-            UploadId: this.uploadId
+            UploadId: id
           }
         })
       );
-    } else if (request.query.uploadId === this.uploadId) {
+    } else if (this.uploadId.includes(request.query.uploadId)) {
+      let arr = [];
```

Comment:
you could do directly instead id delcaring array sperately
```typescript
let arr = Object.values(...)
```

Reply:
changed